biz.aQute.bndlib/src/aQute/bnd/component/AnnotationReader.java

```diff
@@ -940,7 +940,15 @@ protected void doComponent(Component comp, Annotation annotation) throws Excepti
 			// We have explicit interfaces set
 			component.service = new TypeRef[x.length];
 			for (int i = 0; i < x.length; i++) {
-				component.service[i] = (TypeRef) x[i];
+				TypeRef typeRef = (TypeRef) x[i];
+				Clazz service = analyzer.findClass(typeRef);
+				if (assignable(clazz, service) == Boolean.FALSE) {
+					analyzer.error("Class %s is not assignable to specified service %s", clazz.getFQN(),
```

Comment:
Perhaps this should just be a warning and not an error. It is also possible that the service type is deeply inherited and not visible to bnd.